### DISCUSSION TITLE
[DOC] Document JSON for default-disks-config

### DISCUSSION THREAD

**Discussion by guettli** [longhorn/longhorn#8972] [General]
## What's the document you plan to update? Why? Please describe

The [documentation "customizing default disk and node config"](https://longhorn.io/docs/1.6.2/nodes-and-volumes/nodes/default-disk-and-node-config/#customizing-default-disks-for-new-nodes) shows this snippet:

```
node.longhorn.io/default-disks-config: 
'[
    { 
        "path":"/mnt/disk1",
        "allowScheduling":true
    },
    {   
        "name":"fast-ssd-disk", 
        "path":"/mnt/disk2",
        "allowScheduling":false,
        "storageReserved":10485760,
        "tags":[
            "ssd",
            "fast"
        ]
    }
]'
```

I am missing in the docs a link to the specification about this data format. Which values are allowed in this json snippet?

Please update the page or add a hyperlink to the schema definition.

Thank you.

**Comment by ejweber** [comment]
I looked into this a bit, and unfortunately, we're simply not well set up to provide a link like you are requesting. The annotation is meant to essentially contain a JSON formatted list of `DiskSpec` objects.

Viewed in Go code:

https://github.com/longhorn/longhorn-manager/blob/b615732bc75d5513a7f3289a667af1f06a55af1d/k8s/pkg/apis/longhorn/v1beta2/node.go#L81-L98
```
type DiskSpec struct {
	// +kubebuilder:validation:Enum=filesystem;block
	// +optional
	Type DiskType `json:"diskType"`
	// +optional
	Path string `json:"path"`
	// +kubebuilder:validation:Enum="";auto;aio
	// +optional
	DiskDriver DiskDriver `json:"diskDriver"`
	// +optional
	AllowScheduling bool `json:"allowScheduling"`
	// +optional
	EvictionRequested bool `json:"evictionRequested"`
	// +optional
	StorageReserved int64 `json:"storageReserved"`
	// +optional
	Tags []string `json:"tags"`
}
```

Viewed in the `nodes.longhorn.io` CRD:

https://github.com/longhorn/longhorn/blob/e759608033d18108d339e7a1ce1764f0fefb6425/chart/templates/crds.yaml#L2369-L2394

However, instead of a map/dictionary of name to DiskSpec object, as seen with a `kubectl get -oyaml`, the annotation takes a list of DiskSpec objects with an additional `name` field. So you can either add an extra `name` field to each `DiskSpec` (as in `"name":"fast-ssd-disk",` from the second item in the doc example), or omit the `name` field (as in the first item in the doc example).

The example already includes all valid fields for a v1 volume that it makes sense to set.

**Comment by guettli** [comment]
@ejweber thank you for your reply. This answers the above question, and leads me to a new follow-up question: https://github.com/longhorn/longhorn/discussions/9057 Do you have an answer for that?
